src/activitys/Activity.ts

```diff
@@ -0,0 +1,169 @@
+import { Metadata, PlayerDeathType } from "main/types";
+import { Combatant } from "../main/combatant";
+import { VideoCategory } from "../main/constants";
+
+/**
+ * Abstract activity class.
+ */
+export default class Activity {
+    protected category: VideoCategory;
+    protected result: boolean;
+    protected combatantMap: Map<string, Combatant>;
+    protected startDate: Date;
+    protected playerDeaths: PlayerDeathType[];
+    protected endDate?: Date;
+    protected zoneID?: number;
+    protected playerGUID?: string;
+
+    constructor(startDate: Date, 
+                category: VideoCategory) 
+    {
+        this.result = false;
+        this.combatantMap = new Map();
+        this.startDate = startDate;
+        this.category = category;
+        this.playerDeaths = [];
+    }
+
+    end(endDate: Date, result: boolean) {
+        this.endDate = endDate;
+        this.result = result;
+    }
+
+    getZoneID() {
+        return this.zoneID;
+    }
+
+    setZoneID(zoneID: number) {
+        this.zoneID = zoneID; 
+    }
+
+    getDuration() {
+        if (!this.endDate) {
+            console.error("[Activity] Failed to get duration of in-progress activity");
```

Comment:
I don't think classes should log much. If you throw an exception, let the catcher handle the logging.